courses/advanced-ada/chapters/generics.rst

```diff
@@ -120,20 +172,94 @@ algorithm:
     end Test_Reverse_Colors_Simple_Pkg;
 
 In this example, we're first instantiating the
-``Simple_Generic_Array_Pkg`` package, thereby creating the ``Color_Pkg``
-package. We then proceed to use this ``Color_Pkg`` package in the
-instantiation of the generic ``Reverse_Array`` procedure. Also, in the
-declaration of the ``My_Colors`` array, we make use of the array type
-definition from the ``Color_Pkg`` package.
+:ada:`Simple_Generic_Array_Pkg` package, thereby creating the
+:ada:`Color_Pkg` package. We then proceed to use this :ada:`Color_Pkg`
+package in the instantiation of the generic :ada:`Reverse_Array`
+procedure. Also, in the declaration of the :ada:`My_Colors` array, we make
+use of the array type definition from the :ada:`Color_Pkg` package.
+
+.. _FormalPackageParametrization:
+
+Formal package parametrization
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Note that we're using partial parametrization for the formal package
+parameter :ada:`P` in the previous example. Partial parametrization makes
+use of :ada:`others => <>` to indicate that the generic declaration takes
+the definitions from the package argument provided in the generic
+instantiation:
+
+.. code:: ada
+
+    with Simple_Generic_Array_Pkg;
+
+    package Show_Partial_Parametrization is
+
+       generic
+          type T is private;
+          with package P is new Simple_Generic_Array_Pkg (T => T, others => <>);
+       procedure Reverse_Array (X : in out P.Array_T);
+
+    end Show_Partial_Parametrization;
+
+For the previous example, the definitions come from the declarations of
+the :ada:`Color_Pkg` package:
+
+A complete parametrization, in constrast, contains the definition of all
+types in the generic declaration. For example:
+
+.. code:: ada
+
+    with Simple_Generic_Array_Pkg;
+
+    package Show_Complete_Parametrization is
+
+       generic
+          type T is private;
+          type Index is range <>;
+          with package P is new Simple_Generic_Array_Pkg (T     => T,
+                                                          Index => Index);
+       procedure Reverse_Array (X : in out P.Array_T);
+
+    end Show_Complete_Parametrization;
+
+Another approach is to take the all definitions from the formal package
```

Comment:
typo: "take the all"

Reply:
Good catch! I've just fixed it on this small PR: https://github.com/AdaCore/learn/pull/182